.github/copilot-instructions.md

```diff
@@ -0,0 +1,158 @@
+# Deephaven MCP Repository Instructions
+
+Always follow these instructions precisely when working in this repository. Only search for additional context if the information here is incomplete or incorrect.
+
+## Working Effectively
+
+Bootstrap and build the repository:
+- `git clone https://github.com/deephaven/deephaven-mcp.git` -- timing varies by network speed and repository size
+- `cd deephaven-mcp`
+- `pip install uv` -- if uv not installed
+- `uv venv .venv -p 3.12` -- takes 0.01 seconds  
+- `source .venv/bin/activate` (Linux/macOS) or `.venv\Scripts\activate` (Windows)
+- `uv pip install ".[dev]"` -- installs all dependencies including dev tools, takes 3 seconds. NEVER CANCEL. Set timeout to 60+ seconds.
+
+For enterprise features, install the Core Plus wheel separately:
+- `./bin/dev_manage_coreplus_client.sh install-wheel --file /path/to/wheel.whl` -- install from local wheel file
+
+Run tests:
+- `uv run pytest tests/test_init.py -v` -- basic smoke test, takes 0.5 seconds
+- `uv run pytest tests/test__*.py -v` -- core tests, takes 2 seconds. NEVER CANCEL.
+- `uv run pytest` -- full test suite (may require additional dependencies)
+
+## MCP Servers
+
+This repository provides two Model Context Protocol (MCP) servers:
+
+**Systems Server** (orchestrates Deephaven Community Core workers):
+- `dh-mcp-systems-server --help` -- shows available options
+- `DH_MCP_CONFIG_FILE=/path/to/config.json dh-mcp-systems-server --transport sse` -- starts on port 8000
+- `DH_MCP_CONFIG_FILE=/path/to/config.json dh-mcp-systems-server` -- stdio mode (default)
+
+**Docs Server** (documentation Q&A):
+- `INKEEP_API_KEY=your-key dh-mcp-docs-server --transport sse` -- starts on port 8001 
+- `INKEEP_API_KEY=your-key dh-mcp-docs-server` -- streamable-http mode (default)
+- Use `PORT=8001` environment variable to change port
+
+## Configuration
+
+**Systems Server Configuration:**
+- Requires `DH_MCP_CONFIG_FILE` environment variable pointing to JSON config file
+- Create a basic config file for testing:
+```bash
+cat > /tmp/deephaven_mcp.json << 'EOF'
+{
+  "community": {
+    "sessions": {
+      "local_test": {
+        "host": "localhost",
+        "port": 10000,
+        "session_type": "python",
+        "auth_type": "anonymous"
+      }
+    }
+  }
+}
+EOF
+```
+
+**Docs Server Configuration:**
```

Comment:
Docs server configuration should also include a link to the production streamable http MCP server.